src/shiv/bootstrap/__init__.py

```diff
@@ -155,15 +169,15 @@ def bootstrap():  # pragma: no cover
         if env.entry_point is not None:
             mod = import_string(env.entry_point)
             try:
-                sys.exit(mod())
+                run(mod)
             except TypeError:
                 # catch "<module> is not callable", which is thrown when the entry point's
                 # callable shares a name with it's parent module
                 # e.g. "from foo.bar import bar; bar()"
-                sys.exit(getattr(mod, env.entry_point.replace(":", ".").split(".")[1])())
+                run(getattr(mod, env.entry_point.replace(":", ".").split(".")[1]))
```

Comment:
The use of `env.entry_point` here is problematic as it's likely already been scrubbed by the call to run on line 172...